src/modules/hunting/apiserver.py

```diff
@@ -0,0 +1,716 @@
+                               headers={'Authorization': 'Bearer ' + self.service_account_token_evidence}, verify=False)
+            self.api_server_evidence = res.content
+            return res.status_code == 200 and res.content != ''
+        except requests.exceptions.ConnectionError:
+            return False
+
+    def get_service_account_token(self):
+        logging.debug(self.event.host)
+        logging.debug('Passive Hunter is attempting to access pod\'s service account token')
+        try:
+            with open('/var/run/secrets/kubernetes.io/serviceaccount/token', 'r') as token:
+                data = token.read()
+                self.service_account_token_evidence = data
+                return True
+        except IOError:  # Couldn't read file
+            return False
+
+    # 2 Pods Methods:
+    def get_pods_list_under_default_namespace(self):
+        try:
+            res = requests.get("https://{host}:{port}/api/v1/namespaces/default/pods".format(host=self.event.host,
+                                port=self.event.port),
+                               headers={'Authorization': 'Bearer ' + self.service_account_token_evidence}, verify=False)
+
+            parsed_response_content = json.loads(res.content.replace('\'', '\"'))
+            for item in parsed_response_content["items"]:
+                name = item["metadata"]["name"].encode('ascii', 'ignore')
+                namespace = item["metadata"]["namespace"].encode('ascii', 'ignore')
+
+                self.namespaces_and_their_pod_names.append({'name': name, 'namespace': namespace})
+
+            return res.status_code == 200
+        except (requests.exceptions.ConnectionError, KeyError):
+            return False
+
+    def get_pods_list_under_all_namespace(self):
+        try:
+            res = requests.get("https://{host}:{port}/api/v1/pods".format(host=self.event.host, port=self.event.port),
+                               headers={'Authorization': 'Bearer ' + self.service_account_token_evidence}, verify=False)
+
+            parsed_response_content = json.loads(res.content.replace('\'', '\"'))
+            for item in parsed_response_content["items"]:
+                name = item["metadata"]["name"].encode('ascii', 'ignore')
+                namespace = item["metadata"]["namespace"].encode('ascii', 'ignore')
+
+                self.namespaces_and_their_pod_names.append({'name': name, 'namespace': namespace})
+
+            return res.status_code == 200
+        except (requests.exceptions.ConnectionError, KeyError):
+            return False
+
+    # 1 Namespace method:
+    def get_all_namespaces(self):
+        try:
+            res = requests.get("https://{host}:{port}/api/v1/namespaces".format(host=self.event.host,
+                                                                                port=self.event.port),
+                               headers={'Authorization': 'Bearer ' + self.service_account_token_evidence},
```

Comment:
Since you use the same auth header all over the place, consider storing it in, say, `self.headers` and then you can just have `headers=self.headers` where it's needed